src/orion/benchmark/task/quadratics.py

```diff
@@ -0,0 +1,505 @@
+""" Defines the Quadratics task, as described in section 4.2 of the ABLR paper.
+
+[1] [Scalable HyperParameter Transfer Learning](
+    https://papers.nips.cc/paper/7917-scalable-hyperparameter-transfer-learning)
+"""
+from functools import partial
+from typing import ClassVar, Type, Union, List, Dict, Any, Tuple, Dict
+import numpy as np
+import math
+from dataclasses import dataclass, fields, replace
+from simple_parsing.helpers.hparams import HyperParameters, uniform
+from orion.benchmark.task.task import Task, compute_identity
+
+from orion.benchmark.task.utils import similarity, distance
+
+from logging import getLogger as get_logger
+from dataclasses import is_dataclass, asdict
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class QuadraticsTaskHParams(HyperParameters):
+    # TODO: In the paper, section 4.2, the function is defined over R^3, but I'm
+    # limiting the bounds to this for now since the y can be enormous otherwise.
+    x0: float = uniform(-100., 100., discrete=False)
+    x1: float = uniform(-100., 100., discrete=False)
+    x2: float = uniform(-100., 100., discrete=False)
+
+
+from typing import Optional
+from dataclasses import field
+
+
+@dataclass
+class QuadraticsTask(Task, HyperParameters):
+    hparams: ClassVar[Type[HyperParameters]] = QuadraticsTaskHParams
+
+    a2: Optional[float] = uniform(0.1, 10.0, default=None)
+    a1: Optional[float] = uniform(0.1, 10.0, default=None)
+    a0: Optional[float] = uniform(0.1, 10.0, default=None)
+
+    task_id: int = 0
+    rng: Optional[np.random.Generator] = field(default=None, repr=False, compare=False)
+    seed: Optional[int] = None
+    max_trials: int = 50
+
+    def __post_init__(self):
+        # super().__post_init__()
+        super().__init__(
+            task_id=self.task_id,
+            rng=self.rng,
+            seed=self.seed,
+            max_trials=self.max_trials,
+        )
+        # Note: rounding to 4 decimals just to prevent some potential issues
+        # related to the string representation of the task later on.
+        self.a2 = np.round(
+            self.rng.uniform(0.1, 10.0) if self.a2 is None else self.a2, 4
+        )
+        self.a1 = np.round(
+            self.rng.uniform(0.1, 10.0) if self.a1 is None else self.a1, 4
+        )
+        self.a0 = np.round(
+            self.rng.uniform(0.1, 10.0) if self.a0 is None else self.a0, 4
+        )
+
+    def with_context(self) -> "QuadraticsTaskWithContext":
+        return QuadraticsTaskWithContext(
+            a2=self.a2, a1=self.a1, a0=self.a0, task_id=self.task_id, rng=self.rng,
+        )
+
+    def __repr__(self) -> str:
+        return f"QuadraticsTask(a0={self.a0}, a1={self.a1}, a2={self.a2})"
+
+    def __call__(
+        self,
+        hp: Union[
```

Comment:
Why does it need to support different types of HPs? The algos should output a specific format with `suggest()`.